### ISSUE TITLE
kernel: support loading 32 bit binaries

### ISSUE THREAD

**Issue by saljam** [ailispaw/docker-root#8]
The current kernel can't run 32 bit binaries, which are still sometimes useful. (e.g. they're used in the boot2gecko toolchain.)

CONFIG_IA32_EMULATION enables this.

**Comment by ailispaw** [comment]
Hi @saljam ,
Thank you for opening an issue.

Could you tell me more detail of your use case?
I would like to check and test it here.

**Comment by saljam** [comment]
Hi @ailispaw, thanks for answering quickly!

With the current docker-root kernel, if I try to run a 32-bit i386 binary (e.g. `clang` from the Android [repository](https://android.googlesource.com/platform/prebuilts/clang/linux-x86/host/3.5/), or any Go program compiled with `GOARCH=386 GOOS=linux go build`) I get the following error:

```
$ ./clang
bash: ./clang: cannot execute binary file: Exec format error
```

The binary is a normal 32-bit ELF file:

```
$ file ./clang
clang: ELF 32-bit LSB executable, Intel 80386, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux.so.2, for GNU/Linux 2.6.15, BuildID[sha1]=69ba21a03efb7f2bc4a26cb83613bc4a111e5ad7, stripped
```

The docker-root kernel itself is 64-bit:

```
$ uname -a
Linux 9c2fb505bced 4.1.6-docker-root #1 SMP Tue Sep 1 02:25:14 UTC 2015 x86_64 GNU/Linux
```

In order to be able to run 32-bit binaries, `CONFIG_IA32_EMULATION` needs to be enabled in the kernel config. It's enabled on the default Debian kernels. I've not tested enabling this yet in docker-root, but I'm 99% sure it's the culprit. :-)

I can send a PR with the new config once I build a kernel to test this out, if you'd like.

Edit: Oh and in my case I need to run that clang because it's part of the Android/FirefoxOS build process.

**Comment by ailispaw** [comment]
I understand that.
Is this situation in the docker-root VM and a container in it as well?

Yes, please send a PR and let me test it, too.

**Comment by saljam** [comment]
Yep, same error message on both docker containers and the VM directly.

```
$ GOARCH=386 GOOS=linux go build hello.go
$ scp hello docker@$(make ip):
$ make ssh
[docker@docker-root ~]$ ./hello 
-bash: ./hello: cannot execute binary file: Exec format error
```

**Comment by saljam** [comment]
Fixed by #9.

Thanks!
